### PR TITLE
Array Search Filter Added

### PR DESCRIPTION
**Pull request by Ketcap**
- [x] DQB-Client now can use array filter

### CODE REVIEW

File: __tests__/ArrayFilter.test.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import { ArrayFilter } from "../src/ArrayFilter";
+import { ArrayFilterOperation } from "../src/FilterOperation";
+
+// Url Parser is not working in this situation
+// it cannot be able to parse "o=Any&p=Tests&v=(o=contains&p=Name&v=1,2,3,4,5,6)"
+// values are overriding each other on parsing phase
+describe("BooleanFilter", () => {
```
Comment (emrekeskinmac) [inline __tests__/ArrayFilter.test.ts]: wrong name

File: src/ArrayFilter.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  public value: string;
+  constructor(params: ArrayFilterParams) {
+    super(params);
+    this.op = params.op;
+    this.value = params.value;
+  }
+  public get innerFilterBuild(): string {
+    // example > "o=Any&p=Tests&v=(o=contains&p=Name&v=1,2,3,4,5,6)"
+    const o = `o=${this.op}`;
+    const p = "p=_"
```
Comment (cyilcode) [inline src/ArrayFilter.ts]: This shouldn't be a const. This filter should be able to form queries towards complex models as well like below
`o=any&p=myCollection&v=(o=in&p=myCollectionPropertyName&v=123)`
notice that i can also search `where myCollection.PropertyName = 123` with the query above.

File: src/FilterOperation.ts [hunk 3]
```diff
@@ -45,3 +46,9 @@ export enum DecimalFilterOperation {
   GreaterThan = "GreaterThan",
   GreaterThanOrEqual = "GreaterThanOrEqual",
 }
+
+export enum ArrayFilterOperation {
+  Any = "Any",
+  In = "In",
```
Comment (cyilcode) [inline src/FilterOperation.ts]: These are actually not array specific. `any` and `all` are.

**Review by emrekeskinmac — APPROVED** [review]

File: package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "dynamic-query-builder-client",
-  "version": "0.1.2",
+  "version": "0.2.0",
```
Comment (cyilcode) [inline package.json:3]: Why have we jumped to `0.2.0` from `0.1.2` ?

**Review by cyilcode — APPROVED** [review]

File: src/FilterOperation.ts (hunk 3 above)
Comment (Ketcap) [reply src/FilterOperation.ts]: My bad it should be all instead in

File: package.json (hunk 4 above)
Comment (Ketcap) [reply package.json:3]: we just added a date filter so its a minor change isn't it ?

File: src/FilterOperation.ts (hunk 3 above)
Comment (Ketcap) [reply src/FilterOperation.ts]: outdated file

**Pull request merged**